clang/lib/Driver/Driver.cpp

```diff
@@ -5399,14 +5416,27 @@ InputInfo Driver::BuildJobsForActionNoCache(
                                                 DependentOffloadKind)}] =
           CurI;
     }
-
-    // Now that we have all the results generated, select the one that should be
-    // returned for the current depending action.
-    std::pair<const Action *, std::string> ActionTC = {
-        A, GetTriplePlusArchString(TC, BoundArch, TargetDeviceOffloadKind)};
-    assert(CachedResults.find(ActionTC) != CachedResults.end() &&
-           "Result does not exist??");
-    Result = CachedResults[ActionTC];
+    // Do a check for a dependency file unbundle for FPGA.  This is out of line
+    // from a regular unbundle, so just create and return the name of the
+    // unbundled file.
+    if (JA->getType() == types::TY_Dependencies &&
+        C.getInputArgs().hasArg(options::OPT_fintelfpga)) {
```

Comment:
Maybe introducing a `types::TY_FPGA_Dependencies` (same `.d` extension), creating the UB action with this new type & changing the type checks accordingly would allow to minimize `OPT_fintelfpga` checks?

Absolutely disregard-able if I'm missing some caveat; the thought is kinda inspired by `types::TY_SYCL_Header`, so I thought I'd throw this in.

Reply:
I like the suggestion.  Added new type to better distinguish the FPGA specific needs.